### ISSUE TITLE
Stockpile not being used: the sequel

### ISSUE THREAD

**Issue by Searthh** [CompletelyFairGames/dwarfcorp#447]
_DwarfCorp Version: 641b9d9 (XNA version) / OS: Windows 10 Home 1703 / 64 bit / 8 GB RAM / Interl i5 7200U @2.50 GHz 2.70 GHz_

I'm so sorry guys... But I think this is a different kind of dwarves-are-not-using -stockpiles-bug. I never got a message that there are not enough stockpiles. Instead, they just idle and not fill the stockpile. In the image you can see a dwarf with a big backpack idling next to the additional stockpiles, which I built on a layer of blue tiles.

![image](https://user-images.githubusercontent.com/32852691/32272979-cc54a252-bf00-11e7-9e8b-ec74166c4863.png)

Eventually, I used God mode to build an additional stockpile in the same area, but when I saw them throwing their stuff through the ground I found out that the stockpile was built 1 level below the previous one (in a space occupied by stone and soil). I'm not sure if this is a separate bug, or even worth fixing since it was built using God Mode. Meanwhile, the craftsdwarf is completely losing its orientation, trying to build a layer of tiles on the bottom floor but climbing snow and running around in seemingly random directions instead.

![image](https://user-images.githubusercontent.com/32852691/32273046-163b454c-bf01-11e7-8fcb-d9e8b8734576.png)

**Comment by mklingen** [comment]
The dwarf running around is the new feature that makes dwarves randomly explore when there's no path. The stockpile not being used seems to be a new bug. Will have to figure out how to reproduce that one.

**Comment by com1clyf3** [comment]
No longer occurring. Closing.

**Comment by com1clyf3** [comment]
I'm wrong. 

This occurred in organic play. Save attached.

My original balloon port stockpile was full, so I built a new stockpile and was rewarded 20 dwarfbux. Noticed that the two crafters who built the  stockpile stood in place next to the stockpile once it  was built. And then, despite having plenty of resources, no one filled up the stockpile.

Save:
[Horn Realm_785_131696225323836184.zip](https://github.com/CompletelyFairGames/dwarfcorp/files/1962998/Horn.Realm_785_131696225323836184.zip)

World:

[Horn Realm.zip](https://github.com/CompletelyFairGames/dwarfcorp/files/1962999/Horn.Realm.zip)


ac0e675c

**Comment by mklingen** [comment]
Okay, thanks for the save data.

**Comment by com1clyf3** [comment]
https://steamcommunity.com/app/252390/discussions/1/1483232961029586464/

**Comment by mklingen** [comment]
That repro happens every time?

**Comment by com1clyf3** [comment]
2/2 tries so far. Take it for a spin.

**Comment by mklingen** [comment]
Okay, will try.

**Comment by mklingen** [comment]
When you mouse over the stockpile while in build or mine mode, what does it say? (It should say "stockpile 0" or similar)

**Comment by com1clyf3** [comment]
I'm not getting info from hovering over stockpile in build or mine.

**Comment by mklingen** [comment]
That's discouraging. That means the zone was never built, or got destroyed immediately. Will investigate.

**Comment by mklingen** [comment]
Sorry, I just can't reproduce. Must be something to do with that world, not the steps you've taken to reproduce it. Will try to revert and then test your old save.

**Comment by mklingen** [comment]
Wow okay, thanks for the save, I fixed it. Forensics were the same as #588 -- but it was caused by a different system I put in place to handle the case where stockpiles had been destroyed by a Gremlin.

1. A dwarf starting building a stockpile *while other dwarfs were mining*.
2. A block was destroyed
3. Every time a block gets destroyed, all the stockpiles are checked to see if all of their voxels have been destroyed.
4. For each empty stockpile, it was removed from the stockpiles list.
5. Since the stockpile is being built, it has no containing voxels, so it is incorrectly removed from the stockpile list.

The effect is that any stockpile you were building while other dwarves were mining would be unused. Fixed in master.
